### PR TITLE
Visualizers Updates, Additions, Fixes

### PR DESCRIPTION
**Pull request by at669**
## Proposed change(s)

- Added Default Visualizers for Actionlib, Diagnostics
- Added preliminary Twist, Wrench message visualizers
- Added Odom visualizer
- Fixes to visualizers using scriptable object (added topic field, fixed null issue for LaserScan, MultiEchoLaserScan, PointCloud, PointCloud2)
- Some convention cleanup

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

[AIRO-470](https://jira.unity3d.com/browse/AIRO-470), [475](https://jira.unity3d.com/browse/AIRO-475), [476](https://jira.unity3d.com/browse/AIRO-476)

### Types of change(s)

- [x] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Passed TargetPlacement tests,, ran [demo.bag](http://wiki.ros.org/rosbag/Tutorials/reading%20msgs%20from%20a%20bag%20file) and other sample rosbags for visual verification. 

### Test Configuration:
- Unity Version: [e.g. Unity 2020.2.0f1]: 2020.3.11
- Unity machine OS + version: [e.g. Windows 10]: MacOS 10.15.7
- ROS machine OS + version: [e.g. Ubuntu 18.04, ROS Noetic]: Ubuntu  20.04, ROS Noetic
- ROS–Unity communication: [e.g. Docker]: VirtualBox Bridged Adapter

## Checklist
- [x] Ensured this PR is up-to-date with the `laurie/VisualizationPackage` branch
- [x] Created this PR to target the `laurie/VisualizationPackage` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/main/CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [Changelog](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/dev/com.unity.robotics.ros-tcp-connector/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/dev/com.unity.robotics.ros-tcp-connector/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate

### CODE REVIEW

File: com.unity.robotics.message-visualizations/Runtime/DefaultVisualizers/Geometry/DefaultVisualizerTwistStamped.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public float thickness = 0.01f;
+        public float lengthScale = 1.0f;
+        public float sphereRadius = 1.0f;
+        public GameObject origin;
+        [SerializeField]
+        Color m_Color;
+
+        public override void Draw(BasicDrawing drawing, MTwistStamped message, MessageMetadata meta)
+        {
+            var orig = origin == null ? Vector3.zero : origin.transform.position;
```
Comment (LaurieCheers-unity) [inline com.unity.robotics.message-visualizations/Runtime/DefaultVisualizers/Geometry/DefaultVisualizerTwistStamped.cs:19]: It occurs to me that my "origin" concept is kinda half-baked. This message has a header, which has a frame_id that identifies the transform (position and orientation) of the twist. So by default we should draw the twist at that position.
But I guess this is fine for now, there's a separate ticket to go through and add tf support to everything.

File: com.unity.robotics.message-visualizations/Runtime/DefaultVisualizers/Sensor/DefaultVisualizerLaserScan.cs [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-using System.Collections;
-using System.Collections.Generic;
 using Unity.Robotics.MessageVisualizers;
 using Unity.Robotics.ROSTCPConnector.ROSGeometry;
 using UnityEngine;
 
 public class DefaultVisualizerLaserScan : DrawingVisualFactory<MLaserScan>
 {
-    public LaserScanVisualizerSettings m_Settings;
+    public LaserScanVisualizerSettings settings;
```
Comment (LaurieCheers-unity) [inline com.unity.robotics.message-visualizations/Runtime/DefaultVisualizers/Sensor/DefaultVisualizerLaserScan.cs:9]: What's the UI flow like for these settings objects now? Last time I tried it I couldn't get it to work.

Comment (at669) [reply com.unity.robotics.message-visualizations/Runtime/DefaultVisualizers/Sensor/DefaultVisualizerLaserScan.cs:9]: yeah, i think the null issue before was stopping most of these from working! so the flow should be similar for other message types, e.g.

1. find the applicable DefaultVisualizer component type, e.g. PointCloud2
2. input the topic in the (newly added) topic field on the component, e.g. `/velodyne_points`
3. play rosbag/stream data, click play
4. check on `Viz` on the appropriate topic
5. once the first message is received, the visualizer component updates with the correct channels, which updates the drawing appropriately now

Screen Shot 2021-06-28 at 11 54 56 AM

File: com.unity.robotics.message-visualizations/Examples~/PickAndPlaceVisualizerExample.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ public override void Start()
     public override void Draw(DebugDraw.Drawing drawing, MMoverServiceResponse message, MessageMetadata meta, Color color, string label)
     {
         int Idx = 1;
-        foreach(MRobotTrajectory trajectory in message.trajectories)
+        foreach (MRobotTrajectory trajectory in message.trajectories)
```
Comment (LaurieCheers-unity) [inline com.unity.robotics.message-visualizations/Examples~/PickAndPlaceVisualizerExample.cs:24]: Old message name

File: com.unity.robotics.message-visualizations/Examples~/PointVisualizerExample.cs [hunk 4]
```diff
@@ -8,28 +8,28 @@
...
-		// you can do that here.
-		string text = $"[{message.x}, {message.y}, {message.z}]";
-		
-		return () =>
-		{
-			// within the body of this function, it acts like a normal Unity OnGUI function.
-			GUILayout.Label(text);
-		};
-	}
+    public override void Draw(MPoint msg, MessageMetadata meta, Color color, string label,
```
Comment (LaurieCheers-unity) [inline com.unity.robotics.message-visualizations/Examples~/PointVisualizerExample.cs:13]: Old message name

Comment (at669) [reply com.unity.robotics.message-visualizations/Examples~/PointVisualizerExample.cs:13]: ah, these were modified by the precommit checks and i didn't notice--is this and the `PrefabTransformVisualizer` safe to delete, given that `DefaultVisualizerPoint` and `DefaultVisualizerTransform` exist + planned future work for custom message tutorialization?

Comment (LaurieCheers-unity) [reply com.unity.robotics.message-visualizations/Examples~/PointVisualizerExample.cs:13]: Oh, I get it. I guess I'd just leave this alone if we're rewriting the tutorial later anyway.

Comment (at669) [reply com.unity.robotics.message-visualizations/Examples~/PointVisualizerExample.cs:13]: sounds good, will leave `PointVisualizerExample`, `PrefabTransformVisualizer`, and `PickAndPlaceVisualizerExample` as they are then!

**Review by LaurieCheers-unity — APPROVED** [review]

**Pull request merged**